Amazon.QLDB.Driver/client/Session.cs

```diff
@@ -237,38 +331,70 @@ internal virtual FetchPageResult FetchPage(string txnId, string nextPageToken)
             {
                 FetchPage = fetchPageRequest,
             };
-            var response = this.SendCommand(request);
+            var response = await this.SendCommand(request, cancellationToken);
             return response.FetchPage;
         }
 
         /// <summary>
-        /// Send a start transaction request to QLDB.
+        /// Send a fetch result request to QLDB, retrieving the next chunk of data for the result.
+        /// </summary>
+        ///
+        /// <param name="txnId">The unique ID of the transaction to execute.</param>
+        /// <param name="nextPageToken">The token that indicates what the next expected page is.</param>
+        ///
+        /// <returns>The result of the <see cref="FetchPageRequest"/>.</returns>
+        internal virtual FetchPageResult FetchPage(string txnId, string nextPageToken)
+        {
+            return this.FetchPageAsync(txnId, nextPageToken).GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        /// Send an asynchronous start transaction request to QLDB.
         /// </summary>
         ///
+        /// <param name="cancellationToken">
+        ///     A cancellation token that can be used by other objects or threads to receive notice of cancellation.
+        /// </param>
+        ///
         /// <returns>The result of the start transaction request.</returns>
-        internal virtual StartTransactionResult StartTransaction()
+        internal virtual async Task<StartTransactionResult> StartTransactionAsync(
+            CancellationToken cancellationToken = default)
         {
             var startTransactionRequest = new StartTransactionRequest();
             var request = new SendCommandRequest
             {
                 StartTransaction = startTransactionRequest,
             };
-            var response = this.SendCommand(request);
+            var response = await this.SendCommand(request, cancellationToken);
             return response.StartTransaction;
         }
 
         /// <summary>
-        /// Send a request to QLDB.
+        /// Send a start transaction request to QLDB.
+        /// </summary>
+        ///
+        /// <returns>The result of the start transaction request.</returns>
+        internal virtual StartTransactionResult StartTransaction()
+        {
+            return this.StartTransactionAsync().GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        /// Send an asynchronous request to QLDB.
         /// </summary>
         ///
         /// <param name="request">The request to send.</param>
+        /// <param name="cancellationToken">
+        ///     A cancellation token that can be used by other objects or threads to receive notice of cancellation.
+        /// </param>
         ///
         /// <returns>The result returned by QLDB for the request.</returns>
-        private SendCommandResponse SendCommand(SendCommandRequest request)
+        private async Task<SendCommandResponse> SendCommand(
```

Comment:
I am reading:

> The name of an async method, by convention, ends with an "Async" suffix.

https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/async/task-asynchronous-programming-model

can you update all async methods to have `async` suffix